src/server.ts

```diff
@@ -53,17 +54,36 @@ export class AttendingServer {
             member.roles.cache.find(role => role.name == queue.name) !== undefined)
     }
 
-    async RemoveMemberFromQueues(member: GuildMember): Promise<number> {
+    async RemoveMemberFromQueues(member: GuildMember, interaction_type: string): Promise<number> {
         let queue_count = 0
         await Promise.all(this.queues.map(queue => {
             if (queue.Has(member)) {
                 queue_count++
-                return queue.Remove(member)
+                return queue.Remove(member, interaction_type)
             }
         }))
         return queue_count
     }
 
+    async RemoveMember(queue_name: string, member: GuildMember, interaction_type: string): Promise<void> {
```

Comment:
Why is this necessary? The `RemoveMemberFromQueues` function already exists. It is what is used for `/leave`. It seems like the slash command and the button should have the same behavior.

Reply:
the /leave command when called where-ever on the server will remove the person from the queue. But there's a leave queue button for each queue. So I don't want a user to press the leave button for one queue and get kicked out of another. Also I'm planning on possibly allowing someone to join more than one queue so it might be useful later